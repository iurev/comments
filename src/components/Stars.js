import React from 'react';
import styled from 'styled-components'
import { Star } from 'styled-icons/boxicons-solid/Star'

const Root = styled.div`
  display: flex;
  flex-direction: row;
`

const StyledStar = styled(Star)`
  width: 20px;
  height: 20px;
  margin-right: 5px;
  path {
    fill: ${p => p.theme.green}
  }
`

const Stars = ({ count }) => (
  <Root>
    {Array(count).fill().map((_, i) => <StyledStar key={i} />)}
  </Root>
)

export default Stars;
