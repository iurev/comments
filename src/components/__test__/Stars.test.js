import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Stars from '../Stars';

it('3', () => {
  const tree = new ShallowRenderer().render(
    <Stars count={3} />
  )
  expect(tree).toMatchSnapshot();
});
