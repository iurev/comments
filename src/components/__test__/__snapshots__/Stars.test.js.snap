// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3 1`] = `
<ForwardRef>
  <ForwardRef />
  <ForwardRef />
  <ForwardRef />
</ForwardRef>
`;
